### PR TITLE
Added Icon Directive

### PR DESCRIPTION
**Pull request by peek4y**
Added a new directive for icons - <code>sm-icon</code>

**Attributes**
 - <code>sm-font-icon</code> : specify the font icon class.
 - <code>sm-icon-size</code> : size of the icon. (in px / em)

### CODE REVIEW

File: src/elements/icon/icon.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  .directive('smIcon', smIcon);
+
+
+  function smIcon() {
+    return {
+      restrict: 'E',
+      replace: true,
+      transclude: true,
+      scope: {
+        fontIcon: '@smFontIcon',
```
Comment (m0t0r) [inline src/elements/icon/icon.js]: Shouldn't we throw if _fontIcon_ is not provided ?

File: src/elements/icon/icon.spec.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  beforeEach(module('semantic.ui.elements.icon'));
+
+  beforeEach(inject(function($rootScope, _$compile_) {
+    $scope = $rootScope.$new();
+    $compile = _$compile_;
+  }));
+
+  it('has to be font icon with classes user & icon', function() {
+    var smIcon = $compile('<sm-icon sm-font-icon="user icon"></sm-icon>')($scope);
```
Comment (m0t0r) [inline src/elements/icon/icon.spec.js]: Why do use _**icon**_ class in the unit tests ?

File: src/elements/icon/icon.js (hunk 1 above)
Comment (peek4y) [inline src/elements/icon/icon.js]: Feels like an overkill to me, to throw.

File: src/elements/icon/icon.spec.js (hunk 2 above)
Comment (peek4y) [inline src/elements/icon/icon.spec.js]: Missed that out. my bad.

File: src/elements/icon/demo/icons.module.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+(function() {
+  'use strict';
+
+  angular
+  .module('semantic.ui.angular.icons', [
+      'ngAria',
```
Comment (m0t0r) [inline src/elements/icon/demo/icons.module.js]: Why do you require _ngAria_ ?

File: src/elements/icon/icon.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  angular
+  .module('semantic.ui.elements.icon', [])
+  .directive('smIcon', smIcon);
+
+
+  function smIcon() {
+    return {
+      restrict: 'E',
+      replace: true,
+      transclude: true,
```
Comment (m0t0r) [inline src/elements/icon/icon.js]: I am not sure we need any transclusion

File: src/elements/icon/icon.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      scope: {
+        fontIcon: '@smFontIcon',
+        iconSize: '@smIconSize'
```
Comment (m0t0r) [inline src/elements/icon/icon.js]: Please, remove `@smIconSize`parameter. Semantic-UI has its own [set of classes](http://semantic-ui.com/elements/icon.html#size) for sizes which we will let them use as needed.

File: src/elements/icon/icon.js (hunk 1 above)
Comment (m0t0r) [inline src/elements/icon/icon.js]: Add a warning for now

File: src/elements/icon/icon.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      },
+      template: getTemplate,
+      link: function(scope, element, attrs, ctrl, transclude) {
+        var node = element[0];
+
+        transclude(scope, function(nodes) {
```
Comment (m0t0r) [inline src/elements/icon/icon.js]: See my comment above

File: src/elements/icon/icon.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+          element.append(nodes);
+        });
+
+        //Adding base icon class, if it is not mentioned explicitly.
+        scope.fontIcon = scope.fontIcon && scope.fontIcon.indexOf('icon') <= -1 ? scope.fontIcon + ' icon' : scope.fontIcon;
```
Comment (m0t0r) [inline src/elements/icon/icon.js]: We do not really need this check. We always expect that user will not provide `icon` class

File: src/elements/icon/icon.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        //Check if the icon size is specified
+        if(scope.iconSize) {
+          element.css('font-size', scope.iconSize);
+        }
+      }
+    }
+  }
+
+  function getTemplate(element, attr) {
+    return attr.smFontIcon ? '<i class="sm-font"></i>' : '';
```
Comment (m0t0r) [inline src/elements/icon/icon.js]: Instead of doing this, add a warning when user did not provide an icon

File: src/elements/icon/icon.js (hunk 7 above)
Comment (caitp) [inline src/elements/icon/icon.js]: Actually, this code doesn't work to begin with --- interpolated isolate scope variables will be overwritten with the original value almost immediately

Comment (caitp) [inline src/elements/icon/icon.js]: Another option is to just include the `icon` class in the directive template

Comment (m0t0r) [inline src/elements/icon/icon.js]: That is what I expected

**Pull request closed without merging**